### PR TITLE
feat: #2056 connect utils

### PR DESCRIPTION
**Pull request by willmcvay**
# Pull request checklist

## Does this close any currently open issues?

fixes: #2056 

**Please check if your PR fulfills the following requirements:**

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [x] Lint (`yarn lint`) has passed locally and any fixes were made for failures
- [x] Test (`yarn test`) has passed locally and any fixes were made for failures

## Pull request type

**Please check the type of change your PR introduces:**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: #2056 

## What is the new behavior?

- Adds Connect Session module to replace Cognito auth
- Refactors Auth on AML app to test and demonstrate new flow

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/connect-session/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Cognito Auth
```
Comment (willmcvay) [inline packages/connect-session/README.md:1]: I need to expand these docs before merging!

File: packages/connect-session/config.example.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "cognitoOAuthUrl": "",
+  "cognitoUserPoolId": ""
```
Comment (willmcvay) [inline packages/connect-session/config.example.json:3]: Userpool can be removed

File: packages/connect-session/tsdx.config.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+module.exports = {
+  rollup(config) {
+    const newConfig = {
+      ...config,
+      output: {
+        ...config.output,
+        globals: {
+          ...config.output.globals,
+          'amazon-cognito-identity-js': 'amazon-cognito-identity-js',
```
Comment (willmcvay) [inline packages/connect-session/tsdx.config.js:11]: Some of these globals can be remove now

File: packages/connect-session/tsdx.config.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+          '@reapit/elements': '../elements',
+        },
+      },
+      plugins: [
+        ...config.plugins,
+        // Have added sass with no emit because elements is a hard dependency. Just needs to be able to
+        // handle the file without throwing
+        scss({
+          output: () => {
+            console.info('Not emitting CSS, plugin to handle the file only')
```
Comment (willmcvay) [inline packages/connect-session/tsdx.config.js:24]: As can sass

File: packages/aml-checklist/src/core/__tests__/private-route-wrapper.tsx [hunk 5]
```diff
@@ -1,27 +1,10 @@
...
+import { PrivateRouteWrapper } from '../private-route-wrapper'
 
 describe('PrivateRouter', () => {
   it('should match a snapshot', () => {
-    expect(toJson(shallow(<PrivateRouteWrapper {...props} />))).toMatchSnapshot()
-  })
-
-  it('should match a snapshot for desktop login', () => {
-    expect(toJson(shallow(<PrivateRouteWrapper {...desktopProps} />))).toMatchSnapshot()
+    expect(toJson(shallow(<PrivateRouteWrapper />))).toMatchSnapshot()
```
Comment (tanphamhaiduong) [inline packages/aml-checklist/src/core/__tests__/private-route-wrapper.tsx:8]: I think no need toJson any more

File: packages/aml-checklist/src/core/__tests__/private-route.tsx [hunk 6]
```diff
@@ -1,47 +1,18 @@
...
 import { shallow, mount } from 'enzyme'
 import toJson from 'enzyme-to-json'
-import { MemoryRouter, Route } from 'react-router'
+import { MemoryRouter } from 'react-router'
 import { PrivateRoute } from '../private-route'
 
 describe('PrivateRouter', () => {
   it('should match a snapshot', () => {
-    expect(toJson(shallow(<PrivateRoute allow="CLIENT" loginType="CLIENT" component={() => null} />))).toMatchSnapshot()
+    expect(toJson(shallow(<PrivateRoute component={() => null} />))).toMatchSnapshot()
```
Comment (tanphamhaiduong) [inline packages/aml-checklist/src/core/__tests__/private-route.tsx:9]: I think no need toJson any more

File: packages/aml-checklist/src/core/connect-session.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import { ReapitConnectBrowserSession } from '@reapit/connect-session'
+// Referencing config direct rather than via window because window.reapit.config not available at compile time
+// and throws a runtime error
+import config from '../../config.json'
+
+// Needs to be a singleton as the class is stateful
+export const reapitConnectBrowserSession = new ReapitConnectBrowserSession({
```
Comment (tanphamhaiduong) [inline packages/aml-checklist/src/core/connect-session.ts:7]: this one can rewrite like this
```
export const createReapitConnectBrowserSession = () => new ReapitConnectBrowserSession({
  connectClientId: window.reapit.config.cognitoClientId,
  connectOAuthUrl: window.reapit.config.cognitoOAuthUrl,
})
```

File: packages/aml-checklist/src/core/private-route-wrapper.tsx [hunk 8]
```diff
@@ -1,70 +1,31 @@
...
 
-export interface PrivateRouteWrapperConnectActions {
-  setRefreshSession: (refreshParams: RefreshParams) => void
-}
-
-export interface PrivateRouteWrapperConnectState {
-  hasSession: boolean
-}
+export const PrivateRouteWrapper: React.FC = ({ children }) => {
+  const session = useReapitConnect(reapitConnectBrowserSession)
```
Comment (tanphamhaiduong) [inline packages/aml-checklist/src/core/private-route-wrapper.tsx:11]:
```
const session = useReapitConnect(createReapitConnectBrowserSession())
```

File: packages/aml-checklist/src/core/connect-session.ts (hunk 7 above)
Comment (willmcvay) [reply packages/aml-checklist/src/core/connect-session.ts:7]: As discussed @tanphamhaiduong the class needs to be instantiated once only as it's stateful. By wrapping in a function we instantiate each time it is required which is needlessly expensive. Agree importing the config not ideal but it's the best I can come up with at the moment without a compromise elsewhere...

**Pull request merged**